contracts/core/EntryPoint.sol

```diff
@@ -100,14 +100,14 @@ contract EntryPoint is IEntryPoint, StakeManager, NonceManager, ReentrancyGuard,
             innerCall = abi.encodeCall(this.innerHandleOp, (callData, opInfo, context));
         }
         bool success;
-        assembly {
+        assembly ("memory-safe") {
             success := call(gas(), address(), 0, add(innerCall, 0x20), mload(innerCall), 0, 32)
             collected := mload(0)
             mstore(0x40, saveFreePtr)
```

Comment:
restoring the "free ptr" is considered safe in this case, since we don't use any memory allocated since we saved this pointer: we allocated `executeUserOp` and then `innerCall`, both used by "call" above, and unused again.

Reply:
Are you sure this is the correct interpretation of the rules? If I understand what they wrote about `"memory-safe"` correctly, they are saying that now Solidity compiler can put variables to the memory, so you can't mess with memory in `assembly`. And here we are, resetting a free memory pointer. 

Also, we are writing to the `0..32` range a lot here. I guess this is fine but if `innerCall` in the future needs to return 64 bytes we need to change that, right?

Reply:
0..32 (and even 0..64) is a "scratch area", and code is allowed freely to write to it.
Yes, this code takes advantage from the fact innerCall return less than 64 bytes. above that you need to allocate buffer (or use multiple returndatacopy calls)